Marlin/src/pins/lpc1768/pins_BTT_SKR.h

```diff
@@ -35,8 +35,15 @@
 #define X_MAX_PIN          P1_28
 #define Y_MIN_PIN          P1_27
 #define Y_MAX_PIN          P1_26
-#define Z_MIN_PIN          P1_25
-#define Z_MAX_PIN          P1_24
+#if EITHER(Z_HOME_DIR == 1, Z_STALL_SENSITIVITY)
+  // Remaping TMC2130 Z1 diag pin (Z_MIN_PIN) to Z_MAX_PIN when upright Z stallguard homing
+  #define Z_MIN_PIN          P1_24
+  #define Z_MAX_PIN          P1_25
+#else
+  // Use normal endstop pins
+  #define Z_MIN_PIN          P1_25
+  #define Z_MAX_PIN          P1_24
+#endif
```

Comment:
Why Z-axis only?
Why only with stallguard?
Why "TMC2130" only?
Why only this board?
Why at all?

Reply:
Why Z-axis only?
Yes 

Why only with stallguard?
Yes

Why "TMC2130" only?
Yes

Why only this board?
Because it is related to the SKR1.3 is manufactured.
The Z Stepstic connector diagnostic pin uses P1_25, the same as Z min Endstop.

If you are trying to use Stallguard to homing the Z axis (x cary assembly) on the upper brackets, as for the prusa i3, the diagnostic pin of the Z Stepstick will not trigger Z_MAX_PIN as it should.

Why at all?
That's all why.
Look SKR 1.3 Board layout
![SKR1 3](https://user-images.githubusercontent.com/33739777/69426430-414b4d00-0cfb-11ea-9849-6f970d87d4fa.jpg)

Reply:
For me it is only for the z axis but, I can also do the pin validation for x and y also in the case where someone uses stallguard and home in the opposite direction.